service/storage_service.cc

```diff
@@ -4501,6 +4501,15 @@ future<sstring> storage_service::wait_for_topology_request_completion(utils::UUI
     co_return sstring();
 }
 
+future<> storage_service::wait_for_topology_not_busy() {
+    auto guard = co_await _group0->client().start_operation(&_group0_as, raft_timeout{});
+    while (_topology_state_machine._topology.is_busy()) {
```

Comment:
May be we should wait for transition state be different from `tablet_migration`? Otherwise if some other operation starts while the code is sleeping it will not detect that the operation completed. Checing for  `tablet_migration` is also not precise. The best would be to set `done` only when we really done. Otherwise the new global request is just a hack to pass some parameters from qp to tablets code through the topology state machine.

Reply:
Good idea. I'll add it as one of potential follow-ups to this PR.

Reply:
Issue created - https://github.com/scylladb/scylladb/issues/19124